File: app/src/main/java/com/lumiyaviewer/lumiya/utils/InlineListEntry.java
package com.lumiyaviewer.lumiya.utils;
import java.util.*;

public interface InlineListEntry<T extends InlineListEntry<T>> {
    InlineList<T> getList();

    T getNext();

    T getPrev();

    void requestEntryRemoval();

    void setList(InlineList<T> inlineList);

    void setNext(T t);

    void setPrev(T t);
}

File: app/src/main/java/com/lumiyaviewer/lumiya/utils/InlineListEntry.kt
package com.lumiyaviewer.lumiya.utils

interface InlineListEntry<T : InlineListEntry<T>> {
    fun getList(): InlineList<T>?
    fun getNext(): T?
    fun getPrev(): T?
    fun requestEntryRemoval()
    fun setList(inlineList: InlineList<T>?)
    fun setNext(t: T?)
    fun setPrev(t: T?)
}